File: backend/app/schemas/doc.py
from pydantic import BaseModel


class DocGenerateRequest(BaseModel):
    ai_enhance: bool = True
    title: str | None = None


class DocResponse(BaseModel):
    id: str
    project_id: str
    doc_type: str
    title: str
    ai_enhanced: bool
    created_at: str
    content_snippet: str


class DocListResponse(BaseModel):
    items: list[DocResponse]
    total: int

File: frontend/src/types/doc.ts
export interface Doc {
  id: string
  project_id: string
  doc_type: string
  title: string
  ai_enhanced: boolean
  created_at: string
  content_snippet?: string
}

export interface DocGenerateRequest {
  title?: string
  ai_enhance?: boolean
}

export interface DocListResponse {
  items: Doc[]
  total: number
}
